### ISSUE TITLE
RePaint Improvements

### ISSUE THREAD

**Issue by nilsleh** [lucidrains/denoising-diffusion-pytorch#317]
@903124 Thanks for the implementation of RePaint! I have been playing around with it, but observed a couple of things that I think I am fixing. However, wanted to discuss this to see whether others can confirm for their tasks. I am using the FFHQ dataset as a toy example here.

As reference I have trained an unconditional `denoising_diffusion_pytorch.GaussianDiffusion` Model and now using the `repaint.py` script for inpainting. With the current implementation, however, I just get sampled images that don't account for the masked ground truth image and I think this is due to some missing arguments.
![inpainted_image_default](https://github.com/lucidrains/denoising-diffusion-pytorch/assets/35272119/78c0b68c-512c-44c1-9fb5-f0142ed64d3b)

When adding the arguments, the samples account for the masked input.
![inpainted_image](https://github.com/lucidrains/denoising-diffusion-pytorch/assets/35272119/58fbccc6-1e8f-4bb0-9c2e-4394616c33d3)

Overall, the generated inpainted samples seem to be a bit blurry, which seems a bit odd, as one could/should just copy the known regions from the masked input and only infill the masked out part. It seems accounted for in the code, but the results don't agree necessarily. 

As another point, I think `repaint.py` has a lot of redundant code like:
- the `Unet` and `Trainer` code, because it is the same as `denoising_diffusion_pytorch.denoising_diffusion_pytorch` and just overwrites the sampling procedure. Maybe removing the redundant code would make the file more concise?
- the `ddim_sampling` is not used

For the inpainting procedure I used this [script](https://gist.github.com/nilsleh/25d77535bbf8e79561ccceae8d5c7354), and I have uploaded the pretrained checkpoint, if you are interested in confirming these observations.

Also tagging @AlejandroSantorum because I saw you ask about Repaint in another issue and thought you might find this useful :)

**Comment by AlejandroSantorum** [comment]
Dear @nilsleh and @lucidrains,

Thank you for tagging me. I have recently found some of the issues that you describe here and I have opened a new PR to address the remaining ones: https://github.com/lucidrains/denoising-diffusion-pytorch/pull/320

I wanted to thank you for uploading an example script of how to use the RePaint method for sampling and inpainting. Really appreciated 💯
